src/coreclr/debug/daccess/dacdbiimpl.cpp

```diff
@@ -6478,6 +6478,10 @@ bool DacHeapWalker::GetSize(TADDR tMT, size_t &size)
             size = AlignLarge(size);
         else
             size = Align(size);
+
+        // If size == 0, it means we have a heap corruption and
+        // we will stuck in an infinite loop, so better fail the call now.
+        ret = size > 0;
```

Comment:
Looks good as is, although would an upper bound check also make sense? Does the GC have an upper bound cap we could check here? I expect those to be more common corruptions than 0, although like it won't cause something as bad as the infinite look in the walk.

Reply:
We could upper bound it by the high alloc mark.